spec/cypress/integration/admin/suppliers/create-lead-provider.js

```diff
@@ -0,0 +1,137 @@
+
+    cy.location("pathname").should(
+      "equal",
+      "/admin/suppliers/new/lead-provider/choose-cip"
+    );
+    cy.get("input[name='lead_provider_form[cip]'][type=radio]").should(
+      "have.length",
+      1
+    );
+    cy.chooseFirstCIPForLeadProvider();
+
+    cy.location("pathname").should(
+      "equal",
+      "/admin/suppliers/new/lead-provider/choose-cohorts"
+    );
+    cy.chooseFirstCohortForLeadProvider();
+
+    cy.location("pathname").should(
+      "equal",
+      "/admin/suppliers/new/lead-provider/review"
+    );
+    cy.get("main").should("contain", leadProviderName);
+    cy.get("main").should("contain", cipName);
+    cy.get("main").should("contain", cohortName);
+    cy.confirmCreateSupplier();
+
+    cy.location("pathname").should(
+      "contain",
+      "/admin/suppliers/new/lead-provider/success"
+    );
+    cy.get("a").contains("manage suppliers").click();
+
+    cy.location("pathname").should("equal", "/admin/suppliers");
+    cy.get("main").should("contain", leadProviderName);
+  });
+
+  it("remembers previous choices", () => {
+    cy.appScenario("admin/suppliers/create_supplier");
+    const leadProviderName = "New lead provider";
+
+    cy.visit("/admin/suppliers");
+    cy.clickCreateSupplierButton();
+
+    cy.chooseSupplierName(leadProviderName);
+    cy.clickBackLink();
+    cy.get("input[type=text]").should("have.value", leadProviderName);
+    cy.clickCommitButton();
+
+    cy.chooseLeadProviderType();
+    cy.clickBackLink();
+    cy.get("input[type=radio][value=lead_provider]").should("be.checked");
+    cy.clickCommitButton();
+
+    cy.chooseFirstCIPForLeadProvider();
+    cy.clickBackLink();
+    cy.get("input[name='lead_provider_form[cip]'][type=radio]").should(
+      "be.checked"
+    );
+    cy.clickCommitButton();
+
+    cy.chooseFirstCohortForLeadProvider();
+    cy.clickBackLink();
+    cy.get("input[name='lead_provider_form[cohorts][]'][type=checkbox]").should(
+      "be.checked"
+    );
+    cy.clickCommitButton();
+
+    cy.get("main").should("contain", leadProviderName);
+    cy.get("main").should("contain", cipName);
+
+    cy.get("main").should("contain", cohortName);
+    cy.confirmCreateSupplier();
+
+    cy.get("a").contains("manage suppliers").click();
+
+    cy.location("pathname").should("equal", "/admin/suppliers");
+    cy.get("main").should("contain", leadProviderName);
+  });
+
+  it("allows changing name choice", () => {
+    cy.appScenario("admin/suppliers/create_supplier");
+    const leadProviderName = "New lead provider";
+
+    cy.visit("/admin/suppliers");
+    cy.clickCreateSupplierButton();
+
+    cy.chooseSupplierName("wrong name");
+    cy.chooseLeadProviderType();
+    cy.chooseFirstCIPForLeadProvider();
+    cy.chooseFirstCohortForLeadProvider();
+
+    cy.get("a").contains("Change name").click();
+    cy.chooseSupplierName(`{selectall}${leadProviderName}`);
+
+    cy.clickCommitButton();
+    cy.clickCommitButton();
+    cy.clickCommitButton();
```

Comment:
sorry, didn't notice this the first time - is this supposed to click three different buttons on different pages? If so, I'd worry that double clicks leading to a failing test if something is slow - maybe testing the URL between clicks could make the test less likely to break

(this is minor though - if it's not already breaking I don't really see why it would start)

Reply:
Yes, that's what it does. Good to know, will avoid in future, and fix this if it breaks